app/controllers/admin/data_update_scripts_controller.rb

```diff
@@ -3,7 +3,13 @@ class DataUpdateScriptsController < Admin::ApplicationController
     layout "admin"
 
     def index
-      @data_update_scripts = DataUpdateScript.all
+      @data_update_scripts = DataUpdateScript.order(run_at: :desc)
+    end
+
+    def force_run
+      response = DataUpdateScript.force_run(params[:id])
```

Comment:
@citizen428 hmm this makes sense. Do you have a suggestion on the best method to tackle this, I was thinking maybe a sidekiq worker?

Reply:
Yes, the re-run should be async.

To update the UI, maybe we could add a `show` action to this controller that returns the status of a single DUS as JSON. Then when you trigger the re-run in the Stimulus controller, you schedule a function with `setTimeOut` or something that periodically checks this endpoint: if the script has finished, update the UI and `clearTimeout`, if it's still in progress reschedule.

But maybe all of this is too complicated and we just schedule the script in Sidekiq, do a full page refresh and put a message in the flash like "Re-run scheduled, this may take a bit"